rpcs3/Emu/Cell/SPURecompiler.cpp

```diff
@@ -7309,6 +7325,55 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 		const auto a = get_vr<f32[4]>(op.ra);
 		const auto b = get_vr<f32[4]>(op.rb);
 
+		if (auto cv = llvm::dyn_cast<llvm::Constant>(b.value))
+		{
+			v128 data = get_const_vector(cv, m_pos, 5000);
+			bool safe_int_compare = true;
+
+			for (u32 i = 0; i < 4; i++)
+			{	
+				const u32 exponent = data._u32[i] & 0x7f800000u;
+
+				if (data._u32[i] > 0x7f7fffffu || !exponent)
+				{
+					// Postive or negative zero, Denormal (treated as zero), Negative constant, or Normalized number with exponent +127
+		 			// Cannot used signed integer compare safely 
+					safe_int_compare = false;
+					break;
+				}
+			}
+
+			if (safe_int_compare)
+			{
+				set_vr(op.rt, sext<s32[4]>(bitcast<s32[4]>(a) > bitcast<s32[4]>(b)));
+				return;
+			}
+		}
+
+		if (auto cv = llvm::dyn_cast<llvm::Constant>(a.value))
+		{
+			v128 data = get_const_vector(cv, m_pos, 5000);
+			bool safe_int_compare = true;
+
+			for (u32 i = 0; i < 4; i++)
+			{	
+				const u32 exponent = data._u32[i] & 0x7f800000u;
+
+				if (data._u32[i] > 0x7f7fffffu || !exponent)
```

Comment:
If the operand is zero/denormal just compare to splat<s32[4>(0).

Reply:
It's unsafe in the case that the other operand is a denormal. It's probably very rare since the SPUs won't be generating denormal numbers, but better to be safe I think.

Reply:
right.